Source/SwiftLintFramework/Rules/UnusedOptionalBindingRule.swift

```diff
@@ -73,11 +76,14 @@ public struct UnusedOptionalBindingRule: ASTRule, ConfigurationProviderRule {
     private func violations(in range: NSRange, of file: File) -> [NSRange] {
         let kinds = SyntaxKind.commentAndStringKinds()
 
-        let underlineOutsideParenthesis = "(?<=[^(]\\s)_(?=\\s[^)])"
-        let underlineInsideParenthesis = "\\((\\s*[_,]\\s*)+\\)"
-        let pattern = underlineOutsideParenthesis + "|" + underlineInsideParenthesis
-        return file.match(pattern: pattern,
-                          excludingSyntaxKinds: kinds,
-                          range: range)
+        let letUnderscore = "let\\s+(_\\s*[=,)])"
+        let letUnderscoreTuple = "let\\s+(\\((\\s*[_,]\\s*)+\\)\\s*=)"
+
+        let matches = file.matchesAndSyntaxKinds(matching: letUnderscore, range: range)
```

Comment:
Can we combine the regex in one? It should be faster than calling `file.matchesAndSyntaxKinds` twice.